deepspeed/runtime/engine.py

```diff
@@ -981,6 +969,10 @@ def _configure_distributed_model(self, model):
                     hasattr(param,
                             'ds_id') for param in self.module.parameters()):
                 self.__check_params(self.module, torch.bfloat16)
+            if self.zero_optimization_stage() == 0 and not self.pipeline_parallelism:
+                raise NotImplementedError(
+                    "When not running ZeRO, BF16 training support is only supported for Pipeline parallelism"
```

Comment:
Hello, I wonder why BF16 is only supported for Pipeline parallelism or ZeRO 1 to ZeRO3, since there is not such limit in prior version.

Reply:
@kisseternity, apologies for the confusion here. This is a new `bf16+Pipeline parallelism` [code path](https://github.com/microsoft/DeepSpeed/blob/master/deepspeed/runtime/bf16_optimizer.py) that was written in the last minute for [BLOOM ](https://huggingface.co/blog/bloom-megatron-deepspeed#bf16optimizer) model training. The existing restrictions in combining with ZeRO are temporary. We plan to harmonize these combinations and eliminate the confusions.

Reply:
> @kisseternity, apologies for the confusion here. This is a new `bf16+Pipeline parallelism` [code path](https://github.com/microsoft/DeepSpeed/blob/master/deepspeed/runtime/bf16_optimizer.py) that was written in the last minute for [BLOOM ](https://huggingface.co/blog/bloom-megatron-deepspeed#bf16optimizer) model training. The existing restrictions in combining with ZeRO are temporary. We plan to harmonize these combinations and eliminate the confusions.

Thanks for replying. In that case, I think bf16 can be used without the  bf16 optimizer or ZeRO as before.

Reply:
Yes, these changes do not affect the previous support for `bf16+ZeRO`.